npm_and_yarn/spec/dependabot/npm_and_yarn/file_updater/npm_lockfile_updater_spec.rb

```diff
@@ -500,4 +500,42 @@
       end
     end
   end
+
+  context "when updating a git source dependency that is not pinned to a hash" do
+    subject { JSON.parse(updated_npm_lock_content) }
+
+    let(:files) { project_dependency_files("npm6/ghpr_no_hash_pinning") }
+    let(:dependency_name) { "discord.js" }
+    let(:version) { "fdad14099779e61cb84dcd1cb2497e0e853a6144" }
+    let(:previous_version) { "ab82cafcde0ee259a32ef14303c1b4a64dea8fae" }
+    let(:requirements) do
+      [{
+        file: "package.json",
+        requirement: "github:discordjs/discord.js",
+        groups: ["dependencies"],
+        source: {
+          type: "git",
+          url: "https://github.com/discordjs/discord.js",
+          branch: nil,
+          ref: "master"
+        }
+      }]
+    end
+    let(:previous_requirements) do
+      [{
+        file: "package.json",
+        requirement: "github:discordjs/discord.js",
+        groups: ["dependencies"],
+        source: {
+          type: "git",
+          url: "https://github.com/discordjs/discord.js",
+          branch: nil,
+          ref: "master"
+        }
+      }]
+    end
+
+    specify { expect(subject["dependencies"]["discord.js"]["version"]).to match(/github:discordjs\/discord.js#[0-9a-z]{40}/) }
+    specify { expect(subject["dependencies"]["discord.js"]["from"]).to eq("github:discordjs/discord.js") }
```

Comment:
Nit, looks like this would run the update twice to check each subject. Thoughts on using a single `it` here instead and check both attributes?